scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,175 @@
+
+This approach needs to be revised for the following reasons:
+
+1.  The vulnerability schema rarely changes, and maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2.  It can be challenging to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process can be complex, as it may require patch release cuts, changes to different branches, and integration between logic that is managed in the master branch, executing other releases binaries.
+3.  The latest offline bundles does not support post `4.5` releases.  Bundle embedding for each release increases the cost to store and consume offline bundles.  The `roxctl` interface and documentation today needs to be clearer on these limitations.
+
+## Decision
+
+We will implement a versioning stream for vulnerability bundles, bumped when we determine that ClairCore's schema has changed.
+
+Initially, two version streams will exist (table 1 below):
+
+| Stream | Source Ref   | Description                                                     |
+|--------|--------------|-----------------------------------------------------------------|
+| dev    | heads/master | This stream tracks the latest vulnerability schema              |
+| v1     | tags/4.5.0   | This stream tracks the schema used by the last StackRox release |
+
+On the subsequent releases (for example, `4.6.0`), one of the two will be carried out:
+
+1.  The schema hasn't changed, so v1 now tracks the new release branch (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.6.0   |
+
+2.  The schema has changed, so v1 stays at the current git ref (`tags/4.5.0`), and a new version, v2, is created to track the next release (`tags/4.6.0`).
+
+   | Stream | Source Ref   |
+   |--------|--------------|
+   | dev    | heads/master |
+   | v1     | tags/4.5.0   |
+   | v2     | tags/4.6.0   |
+
+This process could be either manual or automated.
+
+### Building versioned bundles
+
+#### Online mode
+
+The GitHub workflow will have a map with "bundle version" as keys and "source reference" as values, similar to Table 1 above.  This map can be stored statically in the matrix definition or as a file in the source repository.
+
+A source reference can be one of the following:
+
+1.  A branch reference, `heads/<branch-name>`, example: `heads/release-4.6`
+2.  A tag reference, `tags/<tag-name>`, example: `tags/4.6.0`.
+3.  A StackRox release, `release/<stackrox-release>`, example: `release/4.6.0`
+
+Both (1.) and (2.) will determine a specific commit to use to build the vulnerability exporter.
```

Comment:
When will (1) determine a specific commit? The latest commit in a branch may change. When is the commit determined?

Reply:
1. will determine a branch reference; Branch references move to different commits as changes are done to the repository. Whenever the workflow runs, the current branch reference will be used.